### PR TITLE
More options for build command

### PR DESCRIPTION
**Pull request by flit**
- `-k`, `--clean` to clean the project before generating and building.
- `-x`, `--stop-on-failure` will abort building on the first failure.
- `-j`, `--jobs` lets you set the number of concurrent build jobs. This is currently only implemented for makefiles.

As part of this, the `Project.build()` and `Builder.build_project()` methods now take `**kwargs` to accept arbitrary builder options.

### CODE REVIEW

File: project_generator/commands/build.py [hunk 1]
```diff
@@ -53,3 +60,10 @@ def setup(subparser):
         type=argparse_string_type(str.lower, False), choices=list(ToolsSupported.TOOLS_DICT.keys()) + list(ToolsSupported.TOOLS_ALIAS.keys()))
     subparser.add_argument(
         "-c", "--copy", action="store_true", help="Copy all files to the exported directory")
+    subparser.add_argument(
+        "-k", "--clean", action="store_true", help="Clean project before building")
```
Comment (0xc0170) [inline project_generator/commands/build.py]: unfortunate that -c is taken, `k` should be fine.

File: project_generator/tools/iar.py [hunk 2]
```diff
@@ -100,7 +100,7 @@ def _set_option(self, settings, value):
 
     def _get_option(self, settings, find_key):
         """ Return index for provided key """
-        # This is used as in IAR template, everything 
+        # This is used as in IAR template, everything
```
Comment (0xc0170) [inline project_generator/tools/iar.py]: I would prefer if this is separate commit , fine for now as the changes are easy to follow

**Comment by 0xc0170** [comment]
I'll merge this along with makefile preprocessor , both for v0.10

File: project_generator/tools/iar.py (hunk 2 above)
Comment (flit) [reply project_generator/tools/iar.py]: The whitespace changes you mean? Yeah, that's always a pain… Sorry I didn't separate them.

**Comment by 0xc0170** [comment]
@flit Looks fine to me, going to v0.10 branch where we can continue

**Pull request merged**